src/base/detail/RangeImpl.hh

```diff
@@ -46,6 +46,21 @@ struct RangeTypeTraits
     }
 };
 
+template<class T, class Enable = void>
+struct EnumWithSize
+{
+    static CELER_CONSTEXPR_FUNCTION bool is_valid(T) { return true; }
```

Comment:
I suppose 'technically' it could check that the value fit in the underlying type (so by default `value < std::numeric_limits<int>::max()` (and yes there would be subtleties about make sure that the right type is used for the comparison to be meaningful).

Reply:
This set of traits classes is instantiated on the enum type to begin with... so no comparison with the maximum integer value is meaningful.